### PR TITLE
Fix OpenShift tests for build and push

### PR DESCRIPTION
**Pull request by ashetty1**
[Work in Progress]
This patch fixes the broken buildconfig functional tests and also will add more tests for the new build and push feature.

### CODE REVIEW

**Comment by ashetty1** [comment]
@cdrage @surajnarwade review please

File: script/test_in_openshift/lib.sh [hunk 1]
```diff
@@ -202,6 +202,7 @@ function convert::kompose_up () {
 }
 
 
+
```
Comment (cdrage) [inline script/test_in_openshift/lib.sh]: random newline

File: script/test_in_openshift/compose-files/buildconfig/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+version: "2"
+
+services:
+ foo:
+  build: "./build"
+  command: sleep 100
+
```
Comment (cdrage) [inline script/test_in_openshift/compose-files/buildconfig/docker-compose.yml]: remove newline

File: script/test_in_openshift/compose-files/buildconfig/build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM busybox:1.26.2
+
+RUN touch /test
+
```
Comment (cdrage) [inline script/test_in_openshift/compose-files/buildconfig/build/Dockerfile]: remove newline

**Comment by cdrage** [comment]
@surajssd  Can you do a quick review / look-over and then merge?

**Comment by surajnarwade** [comment]
@ashetty1 , small change
change route from `xip.io` to `nip.io` 
otherwise, LGTM :+1:

**Comment by cdrage** [comment]
Ahhh, this is why the build is failing! See: https://github.com/kubernetes/kompose/issues/752

Please fix the small issue @ashetty1 and then we can merge this in :) thanks!

**Comment by cdrage** [comment]
Thanks @ashetty1 ! Let's see if this new build passes :+1:

**Pull request merged**